pkg/scale/encode.go

```diff
@@ -140,16 +192,19 @@ func (es *encodeState) encodeCustomPrimitive(in interface{}) (err error) {
 	case reflect.Uint64:
 		in = reflect.ValueOf(in).Convert(reflect.TypeOf(uint64(0))).Interface()
 	default:
-		err = fmt.Errorf("unsupported type for custom primitive: %T", in)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedType, in)
 		return
 	}
 	err = es.marshal(in)
+	if err != nil {
+		err = fmt.Errorf("marshalling: %w", err)
+	}
 	return
 }
 
 func (es *encodeState) encodeResult(res Result) (err error) {
 	if !res.IsSet() {
-		err = fmt.Errorf("Result is not set: %+v", res)
+		err = fmt.Errorf("%w: %+v", ErrEncodeResult, res)
```

Comment:
I don't think you created a sentinel error for this error with message `fmt.Errorf("Result is not set: %+v", res)`

Reply:
⬆️ this, we should have an error `ErrResultIsNotSet`